dask_sql/physical/utils/filter.py

```diff
@@ -388,6 +389,18 @@ def _regenerate_collection(
         regen_kwargs = self.creation_info.get("kwargs", {}).copy()
         regen_kwargs = {k: v for k, v in self.creation_info.get("kwargs", {}).items()}
         regen_kwargs.update((new_kwargs or {}).get(self.layer.output, {}))
+
+        if "read_parquet" in str(func):
+            for i in range(len(regen_kwargs["filters"])):
+                new_filters = []
+                for f in regen_kwargs["filters"][i]:
+                    if isinstance(f[2], np.datetime64):
```

Comment:
A small oversight on my part, but this check should be a bit more careful:
```suggestion
                    if len(f) == 3 and isinstance(f[2], np.datetime64):
```